src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AuthorizationApiServiceImpl.scala

```diff
@@ -64,23 +79,39 @@ final case class AuthorizationApiServiceImpl(
     }
   }
 
+  private def getTenantIdOr(
+    selfcareId: UUID
+  )(alternative: => Future[UUID])(implicit contexts: Seq[(String, String)]): Future[UUID] =
+    tenantManagement.getBySelfcareId(selfcareId).map(_.id).recoverWith {
+      case ex if TenantManagementService.is404(ex) => alternative
+    }
+
+  private def upsertTenantBySelfcareId(selfcareId: UUID)(implicit contexts: Seq[(String, String)]): Future[UUID] = for {
+    partyInstitution <- partyProcess.getInstitution(selfcareId)
+    tenantId         <- tenantProcess
+      .selfcareUpsertTenant(partyInstitution.origin, partyInstitution.externalId)(partyInstitution.id)
```

Comment:
I'm totally not sure about this repacking. `partyInstitution` has both origin and originId, so I'm not sure about which to use. I'll ask for details.